contracts/src/v0.8/dev/OptimismSequencerUptimeFeed.sol

```diff
@@ -0,0 +1,331 @@
+    return "OptimismSequencerUptimeFeed 1.0.0";
+  }
+
+  /// @return L1 sender address
+  function l1Sender() public view virtual returns (address) {
+    return s_l1Sender;
+  }
+
+  /**
+   * @notice Set the allowed L1 sender for this contract to a new L1 sender
+   * @dev Can be disabled by setting the L1 sender as `address(0)`. Accessible only by owner.
+   * @param to new L1 sender that will be allowed to call `updateStatus` on this contract
+   */
+  function transferL1Sender(address to) external virtual onlyOwner {
+    setL1Sender(to);
+  }
+
+  /// @notice internal method that stores the L1 sender
+  function setL1Sender(address to) private {
+    address from = s_l1Sender;
+    if (from != to) {
+      s_l1Sender = to;
+      emit L1SenderTransferred(from, to);
+    }
+  }
+
+  /**
+   * @notice Messages sent by the stored L1 sender will arrive on L2 with this
+   *  address as the `msg.sender`
+   * @return L2-aliased form of the L1 sender address
+   */
+  function aliasedL1MessageSender() public view returns (address) {
+    return AddressAliasHelper.applyL1ToL2Alias(l1Sender());
+  }
+
+  /**
+   * @dev Returns an AggregatorV2V3Interface compatible answer from status flag
+   *
+   * @param status The status flag to convert to an aggregator-compatible answer
+   */
+  function getStatusAnswer(bool status) private pure returns (int256) {
+    return status ? int256(1) : int256(0);
+  }
+
+  /**
+   * @notice Raise or lower the flag on the stored Flags contract.
+   */
+  function forwardStatusToFlags(bool status) private {
+    if (status) {
+      FLAGS.raiseFlag(FLAG_L2_SEQ_OFFLINE);
+    } else {
+      FLAGS.lowerFlag(FLAG_L2_SEQ_OFFLINE);
+    }
+  }
+
+  /**
+   * @notice Helper function to record a round and set the latest feed state.
+   *
+   * @param roundId The round ID to record
+   * @param status Sequencer status
+   * @param timestamp Block timestamp of status update
+   */
+  function recordRound(
+    uint80 roundId,
+    bool status,
+    uint64 timestamp
+  ) private {
+    Round memory nextRound = Round(status, timestamp);
+    FeedState memory feedState = FeedState(roundId, status, timestamp);
+
+    s_rounds[roundId] = nextRound;
+    s_feedState = feedState;
+
+    emit NewRound(roundId, msg.sender, timestamp);
+    emit AnswerUpdated(getStatusAnswer(status), roundId, timestamp);
+  }
+
+  /**
+   * @notice Record a new status and timestamp if it has changed since the last round.
+   * @dev This function will revert if not called from `l1Sender` via the L1->L2 messenger.
+   *
+   * @param status Sequencer status
+   * @param timestamp Block timestamp of status update
+   */
+  function updateStatus(bool status, uint64 timestamp) external override {
+    FeedState memory feedState = s_feedState;
+    requireInitialized(feedState.latestRoundId);
+    if (
+      msg.sender != address(s_l2CrossDomainMessenger) &&
+      s_l2CrossDomainMessenger.xDomainMessageSender() != aliasedL1MessageSender()
```

Comment:
The `aliasedL1MessageSender` was using Arbitrum's aliasing helper, is the current Optimism compatible with this?